### PR TITLE
[#WIP] INT-20: dashboard skunkworks project

### PR DESCRIPTION
**Pull request by imlucas**
See #1 for full blow by blow.

Wercker all set up now so if there is a commit or pull request into dev, it'll run `npm run deploy` (check + test + update [mongodb-js.github.io](http://mongodb-js.github.io/)).

[public wercker project page](https://app.wercker.com/#applications/550092ba210ef9cf6a002fe1)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    if (local && !e.ctrlKey && !e.shiftKey && !e.altKey && !e.metaKey) {
+      e.preventDefault();
+      window.app.navigate(aTag.pathname + aTag.search);
+    }
+  }
+});
+
+
+
+var App = AmpersandState.extend({
```
Comment (imlucas) [inline src/index.js:55]: @todo: Switch to using [ampersand-app](http://npm.io/ampersand-app)

File: src/index.less [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+@import "bootstrap";
+@import "octicons";
+
+@import "./less/variables.less";
+@import "./less/xl.less";
+
+/**
+ * ## Components
+ */
+@import "./less/dropdown-menu.less";
```
Comment (imlucas) [inline src/index.less]: Worth moving all of these bootstrap mods and extensions to their own module?

File: src/less/page-header.less [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.page-header {
+  // font-size: 24px;
```
Comment (imlucas) [inline src/less/page-header.less]: Remove commented out

File: src/less/panel.less [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+.panel {
+  .list-group {
+    margin-bottom: 0;
+
+    .list-group-item {
+      border-width: 1px 0;
+      border-radius: 0;
+    }
+    // &:first-child {
```
Comment (imlucas) [inline src/less/panel.less]: Remove commented out

File: src/less/sidebar.less [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  z-index: 10;
+  -webkit-transform: translateZ(0);
+
+  h1 {
+    font-size: 21px;
+    padding-left: 10px;
+  }
+  .facet-name {
+    .h5;
+    // font-weight: bold;
```
Comment (imlucas) [inline src/less/sidebar.less]: Remove commented out

File: src/less/sidebar.less [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  .form {
+    padding: 0px 15px 0px 15px;
+    .form-control {
+      height: 24px;
+      line-height: 24px;
+    }
+  }
+}
+
+.sidebar-collapsed {
```
Comment (imlucas) [inline src/less/sidebar.less]: Remove

File: src/less/variables.less [hunk 7]
```diff
@@ -0,0 +1,829 @@
+//
```
Comment (imlucas) [inline src/less/variables.less:1]: Remove unused/commented out

File: src/models/memory-sync-mixin.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+// Replaces mongodb-dyno-fixture.
```
Comment (imlucas) [inline src/models/memory-sync-mixin.js:1]: @todo move to module

File: src/views/repo-grid-item.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
+var AmpersandView = require('ampersand-view');
+var $ = require('jquery');
+var popoverTpl = require('./repo-popover.jade');
+
+require('bootstrap/js/tooltip');
+require('bootstrap/js/popover');
+
+module.exports = AmpersandView.extend({
```
Comment (imlucas) [inline src/views/repo-grid-item.js]: @todo Share common bindings with list view.

File: src/views/repo-popover.jade [hunk 10]
```diff
@@ -0,0 +1,24 @@
+p #{description}
+
+table.table.table-condensed(style='margin-bottom: 0;')
```
Comment (imlucas) [inline src/views/repo-popover.jade]: @todo move inline styles to less

File: wercker.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (imlucas) [inline wercker.yml:1]: @todo web hooks to notify gitter/datadog

File: gulpfile.js [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+  util = require('util');
+
+gulp.task('default', ['develop']);
+
+/**
+ * Helper for catching error events on vinyl-source-stream's and showing
+ * a nice native notification and printing a cleaner error message to
+ * the console.
+ */
+function notify(titlePrefix) {
```
Comment (imlucas) [inline gulpfile.js:32]: @todo Should live in mj

File: gulpfile.js [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+    ignore: pkg['dependency-check'].ignore
+  };
+
+  function filterIgnored(results) {
+    return results.filter(function(name) {
+      return opts.ignore.indexOf(name) === -1;
+    });
+  }
+
+  dc(opts, function(err, data) {
```
Comment (imlucas) [inline gulpfile.js:68]: @todo should live in mj

**Comment by rueckstiess** [comment]
Recently I've been putting all `.jade` files into a `templates` folder on the main level. What do you think of this structure: 

```
/models
  foo.js            // Foo model/state
/views
  foo.js            // FooView
/templates
  foo.jade          // main foo template
  foo-popup.jade    // additional templates used in FooView
  foo-form.jade
```

**Comment by rueckstiess** [comment]
Instead of `/less`, call it `/styles` to keep it independent of the language used? Might end up having native `.css` in there.

**Comment by rueckstiess** [comment]
https://github.com/mongodb-js/mongodb-js.github.io/blob/1_org-page/src/models/repo.js#L76

`xhr` not defined, also line 87.

**Comment by rueckstiess** [comment]
Do github repos have a "maintainer" field or something like that? Would be good to see on the dashboard who to talk to if something's not right.

**Comment by imlucas** [comment]
@rueckstiess 

> Do github repos have a "maintainer" field or something like that? Would be good to see on the dashboard who to talk to if something's not right.

How about a direct link to the gitter room for the repo ("Ask a question" button)?

**Comment by imlucas** [comment]
@rueckstiess 

> Recently I've been putting all .jade files into a templates folder on the main level. What do you think of this structure:

Then you work yourself into the `template: require('../../../../detail.jade`)` problem...

How about your idea from a dyno PR way back?  Just keep it all next to each other:

```
views/
    foo.jade
    foo.js
    foo.less
```

**Pull request merged**